### ISSUE TITLE
Add pause/resume methods to Animation baseclass

### ISSUE THREAD

**Issue by tacaswell** [matplotlib/matplotlib#17689]
As pointed out in https://github.com/matplotlib/matplotlib/issues/17685 while you can pause an animation by reaching in and calling methods on the event source, when using `Animation` with blitting that is not enough, you also have to flip the animated artists back to non-animated.  This code (lightly modified from @Luluser question) is a proof-of-concept.

```python
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.animation as animation
from matplotlib.widgets import Button


class PlotAnimation:
    def __init__(self):
        self.fig, self.ax = plt.subplots()
        self.x = np.linspace(-10, 10, 1000)
        self.N = 200
        self.interv = 50
        self.n0 = (
            1.0
            / (4 * np.pi * 2e-4 * 0.1) ** 0.5
            * np.exp(-self.x ** 2 / (4 * 2e-4 * 0.1))
        )
        (self.p,) = self.ax.plot(self.x, self.n0)
        self.anim_running = True
        self.Myanimation = animation.FuncAnimation(
            self.fig, self.update, frames=self.N, interval=self.interv, blit=True
        )

    def update(self, i):
        self.n0 += i / 100 % 5
        self.p.set_ydata(self.n0 % 20)
        return (self.p,)

    def animate(self):
        pause_ax = self.fig.add_axes((0.7, 0.025, 0.1, 0.04))
        pause_button = Button(pause_ax, "pause", hovercolor="0.975")
        pause_button.on_clicked(self._pause)
        plt.show()

    def _pause(self, event):
        if self.anim_running:
            self.Myanimation.event_source.stop()
            self.p.set_animated(False)
            self.anim_running = False
            self.fig.canvas.draw_idle()
        else:
            self.p.set_animated(True)
            self.Myanimation.event_source.start()
            self.anim_running = True


animated_plot = PlotAnimation()
animated_plot.animate()
```

This code should be generalized and used to put a `pause` and `resume` methods on `Animation`.

**Comment by tacaswell** [comment]
For the tags:
 - good first issue because there is very little API design here and only working with a limited slice of the library
 - medium difficulty because you have to read and understand the Animation module.  This is probably not a project for someone just learning Python.

**Comment by DCtheTall** [comment]
Hey, I am interested in trying this out! I can probably get to it at some point in the coming week when I have some free time.

**Comment by tacaswell** [comment]
Great!  We don't have a formal assignment process pre-PR but looking forward to seeing it!

**Comment by deep529** [comment]
Hi,

From the above pasted code, this is the solution I came up with.
This will be implemented in Animation Base Class.

<details><summary>Method to Pause</summary>

- Stop the event using `self.event_source.stop()`
- Set the artist to non-animated (Animation class has access to `fig` object) using
  ```python
  for ax in self.fig.get_axes():
      for line in ax.get_lines():
          line.set_animated(False)
  ```
</details>

<details><summary>Method to Resume</summary>

- Set the artist to animated (Animation class has access to `fig` object) using
  ```python
  for ax in self.fig.get_axes():
      for line in ax.get_lines():
          line.set_animated(True)
  ```
- Start the event using `self.event_source.start()`
</details>

I tried above solution and it worked as expected.
@tacaswell, Let me know if there is any mistake / improvement possible.

@DCtheTall, I am sorry, I could not found other issues suitable to my knowledge. You can work on this issue, if the my solution is not correct. (Hope you don't mind.)

**Comment by DCtheTall** [comment]
@deep529 I am not an owner of this repository, I just volunteered for this issue.

**Comment by jklymak** [comment]
@deep529 Contributions are always welcome.  However, if someone else has started working on something, its generally good form to ask before submitting a PR that solves the same problem.  So in this case, lets give @DCtheTall a few days to open a PR.  If he does, then perhaps you can help by reviewing.  Of course, if he doesn't get to it in a few days, feel free to re-ping and we can reconsider.  Thanks!

**Comment by deep529** [comment]
@jklymak, totally agree with your point.

But I jumped in because, he mentioned **he will start working in the next week when he will get free time.**
That's why I thought it is okay for me to try this out.

I did not want to hurt/discourage anyone. Really sorry if I did.

**Comment by DCtheTall** [comment]
@deep529 thanks for being understanding!

I'm getting to it today. I had replied to the issue on a Sunday night and didn't want to overpromise, which is why I gave the flexible timeframe.

**Comment by DCtheTall** [comment]
@tacaswell I built the repo on my Debian machine using the instructions [here](https://github.com/matplotlib/matplotlib/blob/master/INSTALL.rst).

I installed matplotlib using `python3 -m pip3 install .`, and am trying to run tests but getting the following error:

```
_______________________ ERROR collecting test session _______________________
../../.local/lib/python3.7/site-packages/_pytest/config/__init__.py:495: in _importconftest
    return self._conftestpath2mod[key]
E   KeyError: PosixPath('/usr/local/google/home/dylancutler/Desktop/matplotlib/lib/matplotlib/tests/conftest.py')

During handling of the above exception, another exception occurred:
../../.local/lib/python3.7/site-packages/py/_path/local.py:704: in pyimport
    __import__(modname)
lib/matplotlib/__init__.py:174: in <module>
    _check_versions()
lib/matplotlib/__init__.py:159: in _check_versions
    from . import ft2font
E   ImportError: cannot import name 'ft2font' from 'matplotlib' (/usr/local/google/home/dylancutler/Desktop/matplotlib/lib/matplotlib/__init__.py)
```

Here is my `setup.cfg`:

```
[egg_info]

[libs]
system_freetype = False
system_qhull = False

[packages]
tests = True
sample_data = True
```

Any idea on if I am missing anything? Lmk if you need more info.

**Comment by DCtheTall** [comment]
@tacaswell friendly ping

**Comment by jklymak** [comment]
You may want to look at https://github.com/matplotlib/matplotlib/pull/17096

**Comment by DCtheTall** [comment]
@jklymak thanks! Followed the instructions in the PR and was able to run tests.

**Comment by dopplershift** [comment]
Closed by #17826
